app/controllers/jobs_controller.rb

```diff
@@ -50,13 +50,13 @@ def enabled
   end
 
   def destroy
-    if @job.started_by?(current_user) || current_user.is_admin?
+    if @job.can_be_stopped_by?(current_user)
       @job.stop!
-
-      head :ok
     else
-      head :forbidden
+      flash[:error] = "You do not have privileges to stop this job."
     end
+
+    redirect_to [@project, @job]
```

Comment:
Does the `JobsController#destroy` method ever get called via AJAX? If so, a `:forbidden` response makes more sense than a redirect.